common/flatpak-dir.c

```diff
@@ -9663,6 +9663,130 @@ flatpak_dir_find_local_related (FlatpakDir *self,
   return g_steal_pointer (&related);
 }
 
+static GDBusProxy *
+get_accounts_dbus_proxy (void)
+{
+  GDBusConnection *conn = NULL;
+  GDBusProxy *proxy = NULL;
+
+  const char *accounts_bus_name = "org.freedesktop.Accounts";
+  const char *accounts_object_path = "/org/freedesktop/Accounts";
+  const char *accounts_interface_name = accounts_bus_name;
+
+  conn = g_bus_get_sync (G_BUS_TYPE_SYSTEM, NULL, NULL);
+  proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
+                                         G_DBUS_PROXY_FLAGS_NONE,
+                                         NULL,
+                                         accounts_bus_name,
+                                         accounts_object_path,
+                                         accounts_interface_name,
+                                         NULL,
+                                         NULL);
+  g_object_unref (conn);
+
+  return proxy;
+}
+
+static char **
+get_locale_subpaths_from_accounts_dbus (GDBusProxy *proxy)
+{
+  const char *accounts_bus_name = "org.freedesktop.Accounts";
+  const char *accounts_interface_name = "org.freedesktop.Accounts.User";
+  char **object_path;
+  GList *langs = NULL;
+  GList *l = NULL;
+  GPtrArray *subpaths = g_ptr_array_new ();
+  int i;
+
+  GVariant *ret;
+  ret = g_dbus_proxy_call_sync (G_DBUS_PROXY (proxy),
+                                "ListCachedUsers",
+                                g_variant_new ("()"),
+                                G_DBUS_CALL_FLAGS_NONE,
+                                -1,
+                                NULL,
+                                NULL);
+  if (ret != NULL)
+    {
+      g_variant_get (ret,
+                     "(^ao)",
+                     &object_path);
+      g_variant_unref (ret);
+    }
+
+  if (object_path != NULL)
+    {
+      for (i = 0; object_path[i] != NULL; i++)
+        {
+          GDBusProxy *accounts_proxy = NULL;
+          GVariant *value = NULL;
+          gsize size;
+          char *lang;
+
+          accounts_proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
+                                                          G_DBUS_PROXY_FLAGS_NONE,
+                                                          NULL,
+                                                          accounts_bus_name,
+                                                          object_path[i],
+                                                          accounts_interface_name,
+                                                          NULL,
+                                                          NULL);
+
+          value = g_dbus_proxy_get_cached_property (accounts_proxy, "Language");
+          if (value != NULL)
+            {
+              size = g_variant_get_size (value);
+              lang = g_strdup (g_variant_get_string (value, &size));
+              langs = g_list_append (langs, lang);
```

Comment:
This will duplicate the languages if the same language is in multiple users.
Also, for me, all users on the system has language == "". I think we should handle that case by falling back to "all languages" (even if some other user has a language set).